torch/csrc/utils/tensor_new.cpp

```diff
@@ -214,6 +217,7 @@ Tensor internal_new_from_data(
 
 #ifdef USE_NUMPY
   if (PyArray_Check(data)) {
+    AT_ASSERTM(!pin_memory, "Can't pin tensor constructed from numpy");
```

Comment:
This is an inappropriate use of ASSERT. A user could have made this mistake, and it wouldn't indicate an internal error from PyTorch. Therefore it should be a CHECK.